### PR TITLE
add for check device is jailbroken && support latest device

### PR DESCRIPTION
**Pull request by Vanson**
add for check device is jailbroken && support latest device

### CODE REVIEW

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 1]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPhone10,6"])    return @"iPhone X";
+    
     // iPod
-    if ([platform isEqualToString:@"iPod1,1"])      return @"iPod Touch 1G";
-    if ([platform isEqualToString:@"iPod2,1"])      return @"iPod Touch 2G";
-    if ([platform isEqualToString:@"iPod3,1"])      return @"iPod Touch 3G";
-    if ([platform isEqualToString:@"iPod4,1"])      return @"iPod Touch 4G";
-    if ([platform isEqualToString:@"iPod5,1"])      return @"iPod Touch 5G";
-    if ([platform isEqualToString:@"iPod7,1"])      return @"iPod Touch 6G";
+    if ([platform isEqualToString:@"iPod1,1"])      return @"iPod Touch (1 Gen)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: It's better to write it as (1st Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 2]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPod2,1"])      return @"iPod Touch (2 Gen)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: It's better to write it as (2nd Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 3]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPod3,1"])      return @"iPod Touch (3 Gen)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: It's better to write it as (3rd Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 4]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPod4,1"])      return @"iPod Touch (4 Gen)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: It's better to write it as (4th Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 5]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPod5,1"])      return @"iPod Touch (5 Gen)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: It's better to write it as (5th Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 6]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPod7,1"])      return @"iPod Touch (6 Gen)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: It's better to write it as (6th Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 7]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    
     // iPad
-    if ([platform isEqualToString:@"iPad1,1"])      return @"iPad 1";
+    if ([platform isEqualToString:@"iPad1,1"])      return @"iPad";
+    if ([platform isEqualToString:@"iPad1,2"])      return @"iPad 3G";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: I can't find references to this version code, do you have one?

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 8]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
     if ([platform isEqualToString:@"iPad2,1"])      return @"iPad 2 (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Let's remove network information even here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 9]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"iPad2,2"])      return @"iPad 2 (GSM)";
+    if ([platform isEqualToString:@"iPad2,2"])      return @"iPad 2";
     if ([platform isEqualToString:@"iPad2,3"])      return @"iPad 2 (CDMA)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 10]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"iPad2,4"])      return @"iPad 2 (32nm)";
+    if ([platform isEqualToString:@"iPad2,4"])      return @"iPad 2";
+    if ([platform isEqualToString:@"iPad2,5"])      return @"iPad Mini (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here and the official style is `mini`

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 11]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad2,6"])      return @"iPad Mini";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 12]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad2,7"])      return @"iPad Mini (GSM+CDMA)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 13]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
     if ([platform isEqualToString:@"iPad3,1"])      return @"iPad 3 (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 14]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"iPad3,2"])      return @"iPad 3 (CDMA)";
-    if ([platform isEqualToString:@"iPad3,3"])      return @"iPad 3 (GSM)";
+    if ([platform isEqualToString:@"iPad3,2"])      return @"iPad 3 (GSM+CDMA)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 15]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad3,3"])      return @"iPad 3";
     if ([platform isEqualToString:@"iPad3,4"])      return @"iPad 4 (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 16]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"iPad3,5"])      return @"iPad 4 (GSM)";
-    if ([platform isEqualToString:@"iPad3,6"])      return @"iPad 4 (CDMA)";
+    if ([platform isEqualToString:@"iPad3,5"])      return @"iPad 4";
+    if ([platform isEqualToString:@"iPad3,6"])      return @"iPad 4 (GSM+CDMA)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 17]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
     if ([platform isEqualToString:@"iPad4,1"])      return @"iPad Air (WiFi)";
     if ([platform isEqualToString:@"iPad4,2"])      return @"iPad Air (Cellular)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 18]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"iPad3,2"])      return @"iPad 3 (CDMA)";
-    if ([platform isEqualToString:@"iPad3,3"])      return @"iPad 3 (GSM)";
+    if ([platform isEqualToString:@"iPad3,2"])      return @"iPad 3 (GSM+CDMA)";
+    if ([platform isEqualToString:@"iPad3,3"])      return @"iPad 3";
     if ([platform isEqualToString:@"iPad3,4"])      return @"iPad 4 (WiFi)";
-    if ([platform isEqualToString:@"iPad3,5"])      return @"iPad 4 (GSM)";
-    if ([platform isEqualToString:@"iPad3,6"])      return @"iPad 4 (CDMA)";
+    if ([platform isEqualToString:@"iPad3,5"])      return @"iPad 4";
+    if ([platform isEqualToString:@"iPad3,6"])      return @"iPad 4 (GSM+CDMA)";
     if ([platform isEqualToString:@"iPad4,1"])      return @"iPad Air (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 19]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"iPad4,9"])      return @"iPad mini 3 (China)";
-    if ([platform isEqualToString:@"iPad5,1"])      return @"iPad mini 4 (WiFi)";
-    if ([platform isEqualToString:@"iPad5,2"])      return @"iPad mini 4 (Cellular)";
-    // iPad Pro 9.7
-    if ([platform isEqualToString:@"iPad6,3"])      return @"iPad Pro 9.7 (WiFi)";
-    if ([platform isEqualToString:@"iPad6,4"])      return @"iPad Pro 9.7 (Cellular)";
-    // iPad Pro 12.9
-    if ([platform isEqualToString:@"iPad6,7"])      return @"iPad Pro 12.9 (WiFi)";
-    if ([platform isEqualToString:@"iPad6,8"])      return @"iPad Pro 12.9 (Cellular)";
+    if ([platform isEqualToString:@"iPad4,4"])      return @"iPad Mini 2 (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 20]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad4,5"])      return @"iPad Mini 2 (Cellular)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 21]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad4,6"])      return @"iPad Mini 2";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 22]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad4,7"])      return @"iPad Mini 3";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 23]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad4,8"])      return @"iPad Mini 3";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 24]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad4,9"])      return @"iPad Mini 3";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 25]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad5,1"])      return @"iPad Mini 4 (WiFi)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 26]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"iPad5,2"])      return @"iPad Mini 4 (LTE)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Same here

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 27]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    
+    if ([platform isEqualToString:@"iPad6,11"])      return @"iPad (5th Gen)";
+    if ([platform isEqualToString:@"iPad6,12"])      return @"iPad (5th Gen)";
+    if ([platform isEqualToString:@"iPad7,1"])      return @"iPad Pro 12.9 (2nd Gen)";
+    if ([platform isEqualToString:@"iPad7,2"])      return @"iPad Pro 12.9 (2nd Gen)";
+    if ([platform isEqualToString:@"iPad7,3"])      return @"iPad Pro 10.5";
+    if ([platform isEqualToString:@"iPad7,4"])      return @"iPad Pro 10.5";
+    
     // Apple TV
     if ([platform isEqualToString:@"AppleTV2,1"])   return @"Apple TV 2G";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: In according to [The iPhone Wiki](https://www.theiphonewiki.com/wiki/Models) is better to name it `Apple TV (2nd Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 28]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
-    if ([platform isEqualToString:@"AppleTV3,1"])   return @"Apple TV 3G";
-    if ([platform isEqualToString:@"AppleTV3,2"])   return @"Apple TV 3G";
-    if ([platform isEqualToString:@"AppleTV5,3"])   return @"Apple TV 4G";
-    // Apple Watch
-    if ([platform isEqualToString:@"Watch1,1"])     return @"Apple Watch 38mm";
-    if ([platform isEqualToString:@"Watch1,2"])     return @"Apple Watch 42mm";
-    // Simulator
+    if ([platform isEqualToString:@"AppleTV3,1"])   return @"Apple TV 3";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: In according to [The iPhone Wiki](https://www.theiphonewiki.com/wiki/Models) is better to name it `Apple TV (3rd Gen)

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 29]
```diff
@@ -55,72 +55,88 @@ + (NSString * _Nonnull)devicePlatformString {
...
+    if ([platform isEqualToString:@"AppleTV3,2"])   return @"Apple TV 3 (2013)";
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: In according to [The iPhone Wiki](https://www.theiphonewiki.com/wiki/Models) is better to name it `Apple TV (3rd Gen)` and we are missing `AppleTV5,3` - `Apple TV (4th Gen)` and `AppleTV6,2` - `Apple TV 4K`

File: Source/Additions/UIKit/UIDevice+BFKit.m [hunk 30]
```diff
@@ -175,6 +191,43 @@ + (BOOL)isSimulator {
...
+#if !(TARGET_IPHONE_SIMULATOR)
+    // Check 1 : existence of files that are common for jailbroken devices
+    if ([[NSFileManager defaultManager] fileExistsAtPath:@"/Applications/Cydia.app"] ||
+        [[NSFileManager defaultManager] fileExistsAtPath:@"/Library/MobileSubstrate/MobileSubstrate.dylib"] ||
+        [[NSFileManager defaultManager] fileExistsAtPath:@"/bin/bash"] ||
+        [[NSFileManager defaultManager] fileExistsAtPath:@"/usr/sbin/sshd"] ||
+        [[NSFileManager defaultManager] fileExistsAtPath:@"/usr/sbin/sshd"] ||
+        [[NSFileManager defaultManager] fileExistsAtPath:@"/private/var/lib/apt/"] ||
+        [[UIApplication sharedApplication] canOpenURL:[NSURL URLWithString:@"cydia://package/com.example.package"]]) {
+                                                       return YES;
```
Comment (FabrizioBrancati) [inline Source/Additions/UIKit/UIDevice+BFKit.m]: Fix the indentation

**Comment by Vanson** [comment]
OK, thanks'.I will fix it.

**Pull request closed without merging**